pyproject.toml

```diff
@@ -60,6 +63,10 @@ dev = [
     "black",
     "ruff",
 
+    # typecheckers
+    "pyright >= 1.1.354,<1.2.0",
+    "mypy >= 1.9.0,<2.0.0",
```

Comment:
I can change it, but it seems to me that package versions should always be fixed, otherwise at some point it may happen that previously valid code suddenly ceases to be so and when developing a new feature you will have to spend time on fixing errors caused by a newer version of the package.

Reply:
Done